eth2/utils/ssz_types/src/bitfield.rs

```diff
@@ -0,0 +1,1114 @@
+use core::marker::PhantomData;
+use serde::de::{Deserialize, Deserializer};
+use serde::ser::{Serialize, Serializer};
+use serde_hex::{encode as hex_encode, PrefixedHexVisitor};
+use ssz::{Decode, Encode};
+use typenum::Unsigned;
+
+/// A marker trait applied to `BitList` and `BitVector` that defines the behaviour of a `Bitfield`.
+pub trait BitfieldBehaviour: Clone {}
+
+/// A marker struct used to declare SSZ `BitList` behaviour on a `Bitfield`.
+///
+/// See the [`Bitfield`](struct.Bitfield.html) docs for usage.
+#[derive(Clone, PartialEq, Debug)]
+pub struct BitList<N> {
```

Comment:
What if we named these marker structs `Variable<N>` and `Fixed<N>`, and then had type aliases for `BitList` and `BitVector` like:

```rust
type BitList<N> = Bitfield<Variable<N>>;
type BitVector<N> = Bitfield<Fixed<N>>;
```

Reply:
(I have done this via dodgy renaming in `lib.rs` on my branch, and I like it)

Reply:
Yeah I like this!

FYI, I experimented with only exposing the type alias, but if we don't expose `Bitfield` we don't get documentation for the `Bitfield` methods (which is basically all of them).

Reply:
Cool, I'm happy for `Bitfield` and `Variable` and `Fixed` to all be exposed

Reply:
We can get away with not exposing `Variable` and `Fixed`, which is appealing to me.

Reply:
Happy with that too